include/swift/AST/Decl.h

```diff
@@ -206,22 +206,56 @@ struct OverloadSignature {
   DeclName Name;
 
   /// The kind of unary operator.
-  UnaryOperatorKind UnaryOperator = UnaryOperatorKind::None;
+  UnaryOperatorKind UnaryOperator;
 
   /// Whether this is an instance member.
-  bool IsInstanceMember = false;
+  unsigned IsInstanceMember : 1;
 
-  /// Whether this is a property.
-  bool IsProperty = false;
+  /// Whether this is a variable.
+  unsigned IsVariable : 1;
+
+  /// Whether this is a function.
+  unsigned IsFunction : 1;
 
   /// Whether this signature is part of a protocol extension.
-  bool InProtocolExtension = false;
+  unsigned InProtocolExtension : 1;
+
+  /// Whether this signature is of a member defined in an extension of a generic
+  /// type.
+  unsigned InExtensionOfGenericType : 1;
+
+  OverloadSignature()
+      : UnaryOperator(UnaryOperatorKind::None), IsInstanceMember(false),
+        IsVariable(false), IsFunction(false), InProtocolExtension(false),
+        InExtensionOfGenericType(false) {}
 };
 
 /// Determine whether two overload signatures conflict.
+///
+/// \param sig1 The overload signature of the first declaration.
+/// \param sig2 The overload signature of the second declaration.
+/// \param skipProtocolExtensionCheck If \c true, members of protocol extensions
+///        will be allowed to conflict with members of protocol declarations.
 bool conflicting(const OverloadSignature& sig1, const OverloadSignature& sig2,
                  bool skipProtocolExtensionCheck = false);
```

Comment:
Can we make this previous form of `conflicting` static? It looks like only the new `conflicting` calls it.

Reply:
@DougGregor We do still call the original `conflicting` function [here](https://github.com/apple/swift/pull/15412/files#diff-32cb5387532d533d00872d725748a4d9L921) in order to initially discount decls with different signatures (before we then go to calculate their overload type) – which means that we'll end up calling the first `conflicting` twice if the signatures conflict.

Jordan raised his concerns over this redundant additional calling [here](https://github.com/apple/swift/pull/15412#discussion_r176475034) and [here](https://github.com/apple/swift/pull/15412#discussion_r176786739) – I wasn't too concerned given the logic is fairly lightweight; but it would be nice to refactor in such a way where that's not the case. Thoughts?

(I notice that `overload_swift4.swift` is failing on Linux; I'll look into that...)

Reply:
Ah, I'd missed the remaining call to `conflicting`. I'm not too concerned about the extra calls, because this is a fairly quick check.